src/Microsoft.AspNet.Mvc.Razor/Compilation/CompilerCache.cs

```diff
@@ -194,18 +201,17 @@ public class CompilerCache : ICompilerCache
             };
         }
 
-        private CompilerCacheEntry PopulateCacheSetContext(ICacheSetContext cacheSetContext)
+        private MemoryCacheEntryOptions GetMemoryCacheEntryOptions(string normalizedPath, string relativePath)
         {
-            var entry = (CompilerCacheEntry)cacheSetContext.State;
-            cacheSetContext.AddExpirationTrigger(_fileProvider.Watch(entry.RelativePath));
+            var options = new MemoryCacheEntryOptions();
+            options.AddExpirationTrigger(_fileProvider.Watch(relativePath));
 
-            var globalImportPaths = ViewHierarchyUtility.GetGlobalImportLocations(cacheSetContext.Key);
+            var globalImportPaths = ViewHierarchyUtility.GetGlobalImportLocations(normalizedPath);
```

Comment:
You should be ok passing in the `relativePath` here (in fact it's probably a bug in that we're using the `normalizedPath` here).